addons/survey/views/survey_templates.xml

```diff
@@ -124,6 +124,7 @@
             <t t-foreach='page.question_ids' t-as='question'>
                 <t t-if="question in answer.predefined_question_ids" t-call="survey.question_container"/>
             </t>
+            <t t-call="survey.survey_progression"/>
```

Comment:
You could set the variables inline in your t-call instead of re-testing the display_mode within the called template.

```
<t t-call="survey.survey_progression">
    <t t-set="ids" t-value="survey.question_ids.ids"/>
    <t t-set="id" t-value="question.id"/>
    <t t-set="answered" t-value="'answered'"/>
</t>
```

And please rename "id" and "ids" to more meaningful things.

Reply:
Maybe even directly compute the percentage and give it to the "dumb" percentage template.
Instead of giving it variables to "do stuff" with and you don't really know what's going to happen.

It will make your various percentage computation more clear also.